compiler/src/iree/compiler/Codegen/Common/CMakeLists.txt

```diff
@@ -115,6 +115,7 @@ iree_cc_library(
     IREELinalgExtDialect
```

Comment:
My recent change wound up the build a bit and I think that's the cause for the CI failures you're hitting. The `TransformDialectInterpreter` now relies on `Codegen::Common` (this) because it is pulling in `UserConfig.cpp` to set user configs. That said, `MaterializeUserConfigs.cpp` should be the only file even referencing user configs at this point, so I would just move `UserConfig.cpp` to the cc_library below and reverse the link between `Common` and `TransformDialectInterpreter`.

Reply:
There is a tangentially related detail here, most likely what we should be doing is requiring a callback function to be provided per-backend that registers all of the required dialects with the TransformInterpreter pass. This way we aren't referencing the NVGPU and AMDGPU dialects in Codegen/Common. I was waiting for some other work to land first though that reworks the interpreter a bit.